datalad/crawler/dbs/base.py

```diff
@@ -85,7 +87,15 @@ def save(self):
         lgr.debug("Writing %s to %s" % (self.__class__.__name__, self._filepath))
         with open(self._filepath, 'w') as f:
             json.dump(db, f, indent=2, sort_keys=True, separators=(',', ': '))
-        self.repo.git_add(self._filepath)  # stage to be committed
+
+        # stage to be committed:
+        if isinstance(self.repo, AnnexRepo):
+            self.repo.add(self._filepath, git=True)
+        elif isinstance(self.repo, GitRepo):
+            self.repo.add(self._filepath)
+        else:
+            raise ValueError("Unknown repo: %s" % self.repo)
```

Comment:
oh -- we would always need to do this now whenever we want to add to git directly?
For the sake of our sanity I would prefer to sacrifice interface clarity a bit and add `git=True` for `GitRepo.add` with assertion inside that it is always True.  Then here (and anywhere else where needed) we would just use `.add(path, git=True)` and live happy lives ;)